apps/costcontrol/test/unit/startup_test.js

```diff
@@ -427,4 +427,73 @@ suite('Application Startup Modes Test Suite >', function() {
       CostControlApp.init();
     }
   );
+
+  test(
+    'DSDS Ensure the FTE will be closed the second time when there are a data' +
+      ' slot change',
+    function(done) {
+      var defaultDataSlotId = 0, newDataSlotId = 1,
+          dataSlot = 'ril.data.defaultServiceId';
+      var applicationMode = 'DATA_USAGE_ONLY';
+      function setupSims() {
+        MockNavigatorSettings.mSettings[dataSlot] = defaultDataSlotId;
+
+        setupCardState({cardState: 'ready'});
+        window.Common = new MockCommon({ });
+        window.ConfigManager = new MockConfigManager({
+          fakeSettings: { fte: true },
+          applicationMode: applicationMode
+        });
+
+        MockNavigatorMozMobileConnections.mAddMobileConnection(
+          {  iccId: '0000000' } ,1);
+        MockNavigatorMozIccManager.addIcc('0000000', {});
+
+        SimManager.reset();
+      }
+
+      function _onftestarted(evt) {
+        // Simulating the last step of the fte, the start app is launched.
+        window.ConfigManager.setOption({ fte: false },
+          function(){
+            CostControlApp._startApp(Common.closeFTE);
+          });
+      }
+
+      setupSims();
+
+      // 3. The last step of the updateUI method after the first FTE is a
+      // hash change.
+      this.sinon.stub(CostControlApp, '_onHashChange', function() {
+
+        // 4. Last step, checking if the FTE is closed correctly after changing
+        // data simcard second time.
+        this.sinon.stub(Common, 'closeFTE', function () {
+          done(function() {
+            window.removeEventListener('ftestarted', _onftestarted);
+          });
+        });
+
+        var currentMode = CostControlApp._getCurrentMode();
```

Comment:
Why do we need to check the currentMode? I think this is not necessary as what you want to test is simple the `closeFTE` function continues to work after the switch change.